packages/clerk-js/src/ui/components/UserProfile/ProfilePage.tsx

```diff
@@ -73,6 +86,7 @@ export const ProfilePage = withCardStateProvider(() => {
           <UserProfileAvatarUploader
             user={user}
             onAvatarChange={uploadAvatar}
+            onAvatarRemove={!(user.profileImageUrl || '').includes('gravatar') ? onAvatarRemove : null}
```

Comment:
Also, we should probably refactor this check a bit as right now, the `ProfilePage` component knows too much about the user avatar's internals. This low-level `.includes('gravatar')` looks suspicious here and it should at least be refactored into a util. Unfortunately, right now its not super easy to add it on the user, so lets do a util for the time being

(this should also be applied [here](https://github.com/clerkinc/javascript/blob/1570dcb4d7404433ac7a310311137691fe6b52ef/packages/clerk-js/src/ui/components/UserProfile/UserProfileAvatarUploader.tsx#L14) 

sorry for not catching that in the other PR!